### ISSUE TITLE
[BUG] Chapter 11: ExponentialDecay uses global step number instead of instance step number for learning rate update

### ISSUE THREAD

**Issue by fkalbe** [ageron/handson-ml2#522]
The `ExponentialDecay` class defined in the chapter 11 notebook initializes an instance variable `self.s` that holds the number of steps. In the `on_epoch_begin()` method, the learning rate is updated, but it currently uses `s` instead of `self.s` when calculating the new learning rate:

```python
class ExponentialDecay(keras.callbacks.Callback):
    def __init__(self, s=40000):
        super().__init__()
        self.s = s

    def on_batch_begin(self, batch, logs=None):
        # Note: the `batch` argument is reset at each epoch
        lr = K.get_value(self.model.optimizer.learning_rate)
        K.set_value(self.model.optimizer.learning_rate, lr * 0.1**(1 / s))

[...]

s = 20 * len(X_train) // 32 # number of steps in 20 epochs (batch size = 32)
exp_decay = ExponentialDecay(s)
history = model.fit(X_train_scaled, y_train, epochs=n_epochs,
                    validation_data=(X_valid_scaled, y_valid),
                    callbacks=[exp_decay])
```

The reason that this doesn't throw an exception is because `s` is defined in the global scope where it was used to initialize the `exp_decay` object. Still, I think the original intention was probably to use `self.s` in `on_batch_begin()`, right?

**Comment by ageron** [comment]
Oh good catch, thanks! I'll fix this now.
